apps/wpcom-block-editor/src/wpcom/features/live-preview.tsx

```diff
@@ -0,0 +1,103 @@
+import { getQueryArg } from '@wordpress/url';
+import { useEffect } from 'react';
+
+/**
+ * Return true if the user is currently previewing a theme.
+ * FIXME: This is copied from Gutenberg, and that should be accessed from the `core/edit-site` store.
+ * @see https://github.com/WordPress/gutenberg/blob/053c8f733c85d80c891fa308b071b9a18e5194e9/packages/edit-site/src/utils/is-previewing-theme.js#L6
+ * @returns {boolean} isPreviewingTheme
+ */
+function isPreviewingTheme() {
+	return getQueryArg( window.location.href, 'wp_theme_preview' ) !== undefined;
+}
+
+/**
+ * Return the theme slug if the user is currently previewing a theme.
+ * FIXME: This is copied from Gutenberg, and that should be accessed from the `core/edit-site` store.
+ * @see https://github.com/WordPress/gutenberg/blob/053c8f733c85d80c891fa308b071b9a18e5194e9/packages/edit-site/src/utils/is-previewing-theme.js#L6
+ * @returns {string|null} currentlyPreviewingTheme
+ */
+function currentlyPreviewingTheme() {
+	if ( isPreviewingTheme() ) {
+		return getQueryArg( window.location.href, 'wp_theme_preview' );
+	}
+	return null;
+}
+
+/**
+ * Sometimes Gutenberg doesn't allow you to re-register the module and throws an error.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let unlock: ( object: any ) => any | undefined;
+try {
+	unlock = __dangerousOptInToUnstableAPIsOnlyForCoreModules(
+		'I know using unstable features means my plugin or theme will inevitably break on the next WordPress release.',
+		'@wordpress/edit-site'
+	).unlock;
+} catch ( error ) {
+	// eslint-disable-next-line no-console
+	console.error( 'Error: Unable to get the unlock api. Reason: %s', error );
+}
+
+const NOTICE_ID = 'wpcom-live-preview/notice';
+
+/**
+ * This is an interim solution to clarify to users that they are currently live previewing a theme.
+ * And this should be moved to jetpack-mu-wpcom.
+ * @see https://github.com/Automattic/wp-calypso/issues/82218
+ */
+const LivePreviewNotice = () => {
+	const { createWarningNotice } = useDispatch( 'core/notices' );
+	const { dashboardLink, previewingTheme } = useSelect( ( select ) => {
+		const { getSettings } = unlock( select( 'core/edit-site' ) );
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		const theme = ( select( 'core' ) as any ).getTheme( currentlyPreviewingTheme() );
+		return {
+			previewingTheme: theme?.name?.rendered || 'new',
+			dashboardLink: getSettings().__experimentalDashboardLink,
+		};
+	}, [] );
+	useEffect( () => {
+		if ( ! isPreviewingTheme() ) {
+			return;
+		}
+		createWarningNotice(
+			sprintf(
+				// translators: %s: theme name
+				__( 'You are currently live-previewing the %s theme.', 'wpcom-live-preview' ),
+				previewingTheme
+			),
+			{
+				id: NOTICE_ID,
+				isDismissible: false,
+				actions: [
+					{
+						label: __( 'Back to Themes', 'wpcom-live-preview' ),
+						url: dashboardLink,
+						variant: 'primary',
```

Comment:
I noticed a minor vertical alignment issue on the button. Can we add styles to this button?

We can fix it by adding `line-height: 24px` or removing `display: block`. Here is a demo:

https://github.com/Automattic/wp-calypso/assets/1881481/7622704e-e478-49d2-8d5c-850a1ea2a4c1

Reply:
>I fixed it 👍

That was not accurate. 
If we pass the `secondary`, the UI will be like the following. This is because Gutenberg [computes](https://github.com/WordPress/gutenberg/blob/053c8f733c85d80c891fa308b071b9a18e5194e9/packages/components/src/notice/index.tsx#L126) it as `link`.

<img width="951" alt="Screenshot 2023-10-06 at 10 21 31" src="https://github.com/Automattic/wp-calypso/assets/5287479/5d8a58c6-7966-48d2-9d89-c646d2af68a8">

Reply:
I see... but I think it's better? Since now it has reduced height, so the user will have more screen room to preview the theme 😄 What do you think?

> I was wondering which of the following I should choose

I think it's OK to leave the `Upgrade` button in `primary` variant so that we encourage upsell.

Reply:
I mean if that's what GB dictates (urls shown as links) then we should follow it 😄

Reply:
I could now make the link look like a button by passing the `noDefaultClasses` prop.
But, yeah, let's follow the Gutenberg standard; I think it'd be good regarding link accessibility. :accessibility:

Reply:
> I noticed a minor vertical alignment issue on the button. Can we add styles to this button?
We can fix it by adding line-height: 24px or removing display: block. Here is a demo:

Thank you for pointing it out 🙂 I don't think we need "alignment" if we choose https://github.com/Automattic/wp-calypso/pull/82629#discussion_r1348153038.
(But I feel like I've seen this kind of alignment issue several times.)